Sources/VLCHTTPUploaderController.m

```diff
@@ -303,22 +303,40 @@ - (void)moveFileFrom:(NSString *)filepath
      * while on iOS we have persistent storage, so move it there */
 #if TARGET_OS_IOS
     NSString *fileName = [filepath lastPathComponent];
-    NSArray *searchPaths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
-    NSString *libraryPath = searchPaths[0];
-    NSString *finalFilePath = [libraryPath stringByAppendingPathComponent:fileName];
+    NSString *libraryPath = [NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject];
+    NSString *uploadPath = [[NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES)
+                            firstObject] stringByAppendingPathComponent:@"Upload"];
+
+    NSString *finalFilePath = [libraryPath
+                               stringByAppendingString:[filepath
+                                                        stringByReplacingOccurrencesOfString:uploadPath
+                                                        withString:@""]];
```

Comment:
can we use a better/ more expressive name than final? Something that describes what we're doing here? If I see it correctly you're getting the relative filepath without the created uploadpath by us correct?